src/app/screens/ledger/addStxAddress/index.tsx

```diff
@@ -0,0 +1,224 @@
+                titleFailed={t('LEDGER_CONNECT.TITLE_FAILED')}
+                textFailed={t('LEDGER_CONNECT.STX_SUBTITLE_FAILED')}
+                imageDefault={ledgerConnectStxIcon}
+                isConnectSuccess={isConnectSuccess}
+                isConnectFailed={isConnectFailed}
+              />
+            )}
+            {currentStepIndex === 1 &&
+              (isConnectFailed ? (
+                <LedgerFailView
+                  title={t('LEDGER_CONNECT.TITLE_FAILED')}
+                  text={t('LEDGER_CONNECT.BTC_SUBTITLE_FAILED')}
+                />
+              ) : (
+                <>
+                  <AddAddressHeaderContainer>
+                    <img src={stxIcon} width={32} height={32} alt="stacks" />
+                    <SelectAssetTitle>{t('LEDGER_ADD_ADDRESS.TITLE_VERIFY_STX')}</SelectAssetTitle>
+                  </AddAddressHeaderContainer>
+                  <AddAddressDetailsContainer>
+                    <SelectAssetText centered>{t('LEDGER_ADD_ADDRESS.SUBTITLE')}</SelectAssetText>
+                    <LedgerAddressComponent
+                      title={t('LEDGER_ADD_ADDRESS.STX')}
+                      address={stacksCredentials?.address}
+                    />
+                  </AddAddressDetailsContainer>
+                  <ConfirmationText>{t('LEDGER_ADD_ADDRESS.CONFIRM_TO_CONTINUE')}</ConfirmationText>
+                </>
+              ))}
+            {currentStepIndex === 2 && (
+              <AddressAddedContainer>
+                <img src={checkCircleIcon} alt="Success" />
+                <SelectAssetTitle>{t('LEDGER_STX_WALLET_ADDED.TITLE')}</SelectAssetTitle>
+                <SelectAssetText centered>{t('LEDGER_STX_WALLET_ADDED.SUBTITLE')}</SelectAssetText>
+              </AddressAddedContainer>
+            )}
+          </OnBoardingContentContainer>
+          <OnBoardingActionsContainer>
+            {currentStepIndex === 0 && (
+              <ActionButton
+                processing={isButtonDisabled}
+                disabled={isButtonDisabled}
+                onPress={checkDeviceConnection}
+                text={t(
+                  isConnectFailed
+                    ? 'LEDGER_IMPORT_TRY_AGAIN_BUTTON'
+                    : 'LEDGER_IMPORT_CONNECT_BUTTON',
+                )}
+              />
+            )}
+            {currentStepIndex === 1 && isConnectFailed && (
+              <ActionButton
+                processing={isButtonDisabled}
+                disabled={isButtonDisabled}
+                onPress={handleTryAgain}
+                text={t('LEDGER_IMPORT_TRY_AGAIN_BUTTON')}
+              />
+            )}
+            {currentStepIndex === 2 && (
+              <ActionButton
+                disabled={isButtonDisabled}
+                processing={isButtonDisabled}
+                onPress={handleWindowClose}
+                text={t('LEDGER_IMPORT_CLOSE_BUTTON')}
+              />
+            )}
+          </OnBoardingActionsContainer>
```

Comment:
have you seen this pattern for multi steps UI? https://github.com/secretkeylabs/xverse-web-extension/blob/develop/src/app/screens/restoreWallet/index.tsx

Reply:
Not yet, though I refactored the steps rendering logic in PR#586 to have a switch-case construction with Steps enum which gives more control over the steps so that you can easily change the step to the one you need if it's not always moving straightforward, here's the example: https://github.com/secretkeylabs/xverse-web-extension/blob/5949c9ee65ed33d2ffe859f832d2e14f6a77b36f/src/app/screens/ledger/importLedgerAccount/index.tsx#L508

Reply:
nice, thanks for cleaning that up!